components/shared/Drawer.tsx

```diff
@@ -15,12 +15,20 @@ interface DrawerProps {
   children: React.ReactNode;
   trigger: React.ReactNode;
   showCloseButton?: boolean;
+  asChild?: boolean;
```

Comment:
I added this prop so that the trigger is the JSX that renders and not the default which is a button.

This was an issue prior to this change because the trigger button we had ended up in another button affecting styling.